Client/src/Navigation/appbar.js

```diff
@@ -217,100 +149,171 @@ return RenderRoles;
+                  <AccountCircleIcon/>
+                </ListItemIcon>
+                <ListItemText>
+                  <Link href="/account">
+                    <Typography component="h6" variant="h6">
+                      My Account
+                    </Typography>
+                  </Link>
+                </ListItemText>          
+              </MenuItem>
+            )}
+            {role.toLowerCase()==='admin' && (
+              <MenuItem onClick={handleClose}>
+                <ListItemIcon>
+                  <SettingsIcon/>
+                </ListItemIcon>
+                <ListItemText>
+                  <Link href="/admin-settings">
+                    <Typography component="h6" variant="h6">
+                      Admin Settings
+                    </Typography>
+                  </Link>
+                </ListItemText>          
+              </MenuItem>
+            )}
+            {role.toLowerCase()==='supporter' && (
+              <MenuItem onClick={handleClose}>
+                <ListItemIcon>
+                  <SettingsIcon/>
+                </ListItemIcon>
+                <ListItemText>
+                  <Link href="/supporter-settings">
+                    <Typography component="h6" variant="h6">
+                      Supporter Settings
+                    </Typography>
+                  </Link>
+                </ListItemText>          
+              </MenuItem>
+            )}
+            {(PossibleRoles.length>1) && (
+              <MenuItem onClick={handleModalOpen}>
+                <ListItemIcon>
+                  <UpdateIcon/>
+                </ListItemIcon>
+                <ListItemText>
+                  <Link>
+                    <Typography component="h6" variant="h6">
+                      Switch Role
+                    </Typography>
+                  </Link>
+                </ListItemText>          
+              </MenuItem>
+            )}
+            <MenuItem onClick={logout}>
+              <ListItemIcon>
+                <ExitToAppIcon/>
+              </ListItemIcon>
+              <ListItemText>
+                <Link href="/login">
+                  <Typography component="h6" variant="h6">
+                    Log Out
+                  </Typography>
+                </Link>
+              </ListItemText>   
+            </MenuItem>
+          </Menu>
+            <Dialog
+            open={openModal}
+            TransitionComponent={Transition}
+            keepMounted
+            onClose={handleModalClose}
+            aria-labelledby="alert-dialog-slide-title"
+            aria-describedby="alert-dialog-slide-description"
+          >
+            <DialogTitle id="alert-dialog-slide-title">{"Choose role"}</DialogTitle>
+            <DialogContent>
+              <DialogContentText id="alert-dialog-slide-description">
+                Given below are the list of the roles that you have been registered as. Select the role you want to switch to.
```

Comment:
(Not related to being mobile responsive)

I think if a user clicks on "Choose Role", they would already know that the following is a list of roles. So, I feel like the line about 'Given below are the list of the roles that you been registered as' is redundant and not useful. Rather, that extra line just makes the UI look a bit not good. Maybe a smaller message such as 

"Roles: " and then the list of roles after a <br> or something like that?

Reply:
I think this is still good to have for users that are newer to site so it is 100% clear what everything does